File: Sources/NockerlDesign/Components/NockerlInfoTip.swift
// NockerlInfoTip: a themed informational popover behind a small ⓘ trigger (
// gate 2), the settings-surface companion (it rides ``NockerlFormSection``'s
// headerAccessory slot). Voice's InfoTip is the INPUT; this is the fresh-spec'd
// system piece:
//
//   - Trigger: a flat ⓘ glyph button. It is INTERACTIVE, so it wears the plain
//     interactive treatment (the icon-interactivity canon: inset styling is
//     reserved for informational, non-interactive marks and never a trigger).
//   - Panel: the `cardSurface2` popover plane, clamped at the `size.container.md`
//     token (Voice's raw 250px, tokenized). Presented via the PLATFORM popover
//     (honor-the-platform: the system anchoring/chrome stays native; the content
//     plane is ours).

import SwiftUI

/// A themed info popover behind a flat ⓘ trigger. Tap toggles; the platform
/// popover handles anchoring + dismissal.
public struct NockerlInfoTip: View {
    private let title: String?
    private let text: String
    private let accessibilityName: String

    @State private var open = false
    @Environment(\.colorScheme) private var colorScheme

    /// Create an info tip.
    /// - Parameters:
    ///   - title: optional short heading above the body.
    ///   - text: the informational body.
    ///   - accessibilityName: the trigger's a11y name (default "More information").
    public init(
        title: String? = nil,
        text: String,
        accessibilityName: String = "More information"
    ) {
        self.title = title
        self.text = text
        self.accessibilityName = accessibilityName
    }

    public var body: some View {
        let palette = NockerlPalette.resolve(colorScheme)

        Button {
            open.toggle()
        } label: {
            Image(systemName: "info.circle")
                .font(.system(size: NockerlFontSize.size12, weight: .medium))
                .foregroundColor(palette.onCardMuted)
        }
        .buttonStyle(.plain)
        .accessibilityLabel(accessibilityName)
        .popover(isPresented: $open) {
            NockerlInfoTipPanel(title: title, text: text)
        }
    }
}

/// The info tip's panel, public so custom hosts (and static previews) can
/// compose the exact same themed content the popover presents.
public struct NockerlInfoTipPanel: View {
    private let title: String?
    private let text: String

    @Environment(\.colorScheme) private var colorScheme

    /// Create the panel.
    /// - Parameters:
    ///   - title: optional short heading above the body.
    ///   - text: the informational body.
    public init(title: String? = nil, text: String) {
        self.title = title
        self.text = text
    }

    public var body: some View {
        let palette = NockerlPalette.resolve(colorScheme)

        VStack(alignment: .leading, spacing: NockerlSpace.space1) {
            if let title {
                Text(title)
                    .font(.nockerl(size: NockerlFontSize.size12, weight: .medium))
                    .foregroundColor(palette.onCard)
            }
            Text(text)
                .font(.nockerl(size: NockerlFontSize.size12, weight: .light))
                .foregroundColor(palette.onCardMuted)
        }
        .padding(NockerlSpace.space3)
        .frame(width: NockerlSize.containerMd, alignment: .leading)
        .background(palette.cardSurface2)
    }
}

File: packages/compose/nockerl-design-components/src/main/kotlin/com/dizyx/nockerl/design/components/NockerlInfoTip.kt
package com.dizyx.nockerl.design.components

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Info
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Popup
import com.dizyx.nockerl.design.tokens.LocalNockerlColors
import com.dizyx.nockerl.design.tokens.NockerlElevation
import com.dizyx.nockerl.design.tokens.NockerlPanelShape
import com.dizyx.nockerl.design.tokens.NockerlSize
import com.dizyx.nockerl.design.tokens.NockerlSurface

/**
 * The **info tip**: a themed informational popover behind a small ⓘ trigger
 * and the settings-surface companion (it rides
 * [NockerlFormSection]'s `headerAccessory` slot). Voice's InfoTip is the input;
 * this is the fresh-spec'd system piece:
 *
 * - **Trigger:** a flat ⓘ glyph [NockerlIconButton], INTERACTIVE, so it wears
 *   the plain interactive treatment (the icon-interactivity canon: inset styling
 *   is reserved for informational, non-interactive marks, never a trigger).
 * - **Panel:** a lifted popover card on the `cardSurface2` plane (L3, the
 *   popover rung of the elevation ladder), panel radius, hairline edge, clamped
 *   at the `size.container.md` token (Voice's raw 250px, tokenized).
 *
 * @param text the informational body.
 * @param modifier outer modifier (wraps the trigger).
 * @param title optional short heading above the body.
 * @param contentDescription the trigger's a11y name (default "More information").
 */
@Composable
fun NockerlInfoTip(
    text: String,
    modifier: Modifier = Modifier,
    title: String? = null,
    contentDescription: String = "More information",
) {
    val colors = LocalNockerlColors.current
    var open by remember { mutableStateOf(false) }

    Box(modifier = modifier) {
        NockerlIconButton(
            icon = Icons.Filled.Info,
            contentDescription = contentDescription,
            onClick = { open = !open },
            tint = colors.onCardMuted,
            size = 28.dp,
            iconSize = 16.dp,
        )
        if (open) {
            Popup(onDismissRequest = { open = false }) {
                NockerlInfoTipPanel(text = text, title = title)
            }
        }
    }
}

/**
 * The info tip's **panel**: the themed popover card [NockerlInfoTip] opens.
 * Public so custom anchoring hosts (and static captures, a [Popup] renders in
 * a separate window with no stable golden frame; the DialogCard precedent) can
 * compose the exact same panel.
 *
 * @param text the informational body.
 * @param modifier outer modifier.
 * @param title optional short heading above the body.
 */
@Composable
fun NockerlInfoTipPanel(
    text: String,
    modifier: Modifier = Modifier,
    title: String? = null,
) {
    val colors = LocalNockerlColors.current
    NockerlSurface(
        modifier = modifier.width(NockerlSize.containerMd),
        shape = NockerlPanelShape,
        // The popover plane: one step above the card (cardSurface2), on the
        // popover rung (L3) with the neutral hairline edge.
        color = colors.cardSurface2,
        elevation = NockerlElevation.Level3,
        border = colors.cardHairline,
    ) {
        Column(
            modifier = Modifier.padding(12.dp),
            verticalArrangement =
                androidx.compose.foundation.layout.Arrangement
                    .spacedBy(4.dp),
        ) {
            if (title != null) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.labelMedium,
                    fontWeight = FontWeight.Medium,
                    color = colors.onCard,
                )
            }
            Text(
                text = text,
                style = MaterialTheme.typography.bodySmall,
                color = colors.onCardMuted,
            )
        }
    }
}
